File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] h) {
        int l = 0;
        int r = h.length - 1;
        int minH = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        while (l < r) {
            minH = Math.min(h[r], h[l]);
            max = Math.max(max, minH * (r - l));
            if (h[r] < h[l]) {
                r--;
            } else {
                l++;
            }
        }
        return max;
    }

}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, h: list[int]) -> int:
        left = 0
        right = len(h)-1
        area = 0
        while left < right:
            min_h = min(h[left], h[right])
            area = max(area, min_h * (right - left))
            if h[left] < h[right]:
                left += 1
            else :
                right -= 1
        return area